src/client/js/client/libraries.js

```diff
@@ -0,0 +1,126 @@
+/*globals define*/
+/*jshint browser: true*/
+/**
+ * @author kecso / https://github.com/kecso
+ */
+define(['js/Constants'], function (CONSTANTS) {
+    'use strict';
+    function gmeLibraries(logger, state, storage, saveFunction) {
```

Comment:
during the library functions we do not receive 'new' nodes that needs to be stored, that is why it is not used here (just the saveFunction which I will rename to saveRoot to make the similarity clear)